interchaintest/feegrant_test.go

```diff
@@ -0,0 +1,530 @@
+	t.Parallel()
+	ctx := context.Background()
+	logger := zaptest.NewLogger(t)
+
+	nv := 1
+	nf := 0
+
+	//In order to have this image locally you'd need to build it with heighliner, e.g.,
+	//from within the local "gaia" directory, run the following command:
+	//../heighliner/heighliner build -c gaia --local -f ../heighliner/chains.yaml
+	// gaiaImage := ibc.DockerImage{
+	// 	Repository: "gaia",
+	// 	Version:    "local",
+	// 	UidGid:     "1025:1025", //the heighliner user string. this isn't exposed on ibctest
+	// }
+
+	// gaiaChainSpec := &ibctestv5.ChainSpec{
+	// 	ChainName:     "gaia",
+	// 	NumValidators: &nv,
+	// 	NumFullNodes:  &nf,
+	// 	ChainConfig: ibc.ChainConfig{
+	// 		Type: "cosmos",
+	// 		Name: "gaia",
+	// 		//ChainID:        "gaia-1", //I believe this will be auto-generated?
+	// 		Images:         []ibc.DockerImage{gaiaImage},
+	// 		Bin:            "gaiad",
+	// 		Bech32Prefix:   "cosmos",
+	// 		Denom:          "uatom",
+	// 		GasPrices:      "0.01uatom",
+	// 		TrustingPeriod: "504h",
+	// 		GasAdjustment:  1.3,
+	// 	}}
+
+	// Chain Factory
+	cf := ibctestv5.NewBuiltinChainFactory(zaptest.NewLogger(t), []*ibctestv5.ChainSpec{
+		{Name: "gaia", ChainName: "gaia", Version: "v7.0.3", NumValidators: &nv, NumFullNodes: &nf},
+		//gaiaChainSpec,
+		{Name: "osmosis", ChainName: "osmosis", Version: "v14.0.1", NumValidators: &nv, NumFullNodes: &nf},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+	gaia, osmosis := chains[0], chains[1]
+
+	// Relayer Factory to construct relayer
+	r := NewRelayerFactory(RelayerConfig{
+		Processor:           relayer.ProcessorEvents,
+		InitialBlockHistory: 100,
+	}).Build(t, nil, "")
+
+	processor.PathProcMessageCollector = make(chan *processor.PathProcessorMessageResp, 50)
+
+	// Prep Interchain
+	const ibcPath = "gaia-osmosis"
+	ic := ibctestv5.NewInterchain().
+		AddChain(gaia).
+		AddChain(osmosis).
+		AddRelayer(r, "relayer").
+		AddLink(ibctestv5.InterchainLink{
+			Chain1:  gaia,
+			Chain2:  osmosis,
+			Relayer: r,
+			Path:    ibcPath,
+		})
+
+	// Reporter/logs
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	client, network := ibctestv5.DockerSetup(t)
+
+	// Build interchain
+	require.NoError(t, ic.Build(ctx, eRep, ibctestv5.InterchainBuildOptions{
+		TestName:  t.Name(),
+		Client:    client,
+		NetworkID: network,
+
+		SkipPathCreation: false,
+	}))
+
+	// Get Channel ID
+	gaiaChans, err := r.GetChannels(ctx, eRep, gaia.Config().ChainID)
+	require.NoError(t, err)
+	gaiaChannel := gaiaChans[0]
+	osmosisChannel := gaiaChans[0].Counterparty
+
+	// Create and Fund User Wallets
+	fundAmount := int64(10_000_000)
+
+	// Tiny amount of funding, not enough to pay for a single TX fee (the GRANTER should be paying the fee)
+	granteeFundAmount := int64(10)
+	granteeKeyPrefix := "grantee1"
+	grantee2KeyPrefix := "grantee2"
+	grantee3KeyPrefix := "grantee3"
+	granterKeyPrefix := "default"
+	coinType := "118"
+
+	mnemonicAny := genMnemonic(t)
```

Comment:
Can these use `BuildRelayerWallet` from `github.com/strangelove-ventures/interchaintest/v7/chain/cosmos` to avoid needing to generate the mnemonics here? You can also use the `additionalGenesisWallets` parameter on the `.AddChain` calls (lines 131 and 132) to avoid having to fund them from the faucet post genesis.

Reply:
the mixture of the 2 should get you both. You can restore the keys from BuildRelayerWallet into the chain.